src/database/entities/provider-configuration.entity.ts

```diff
@@ -0,0 +1,123 @@
+// src/database/entities/provider-configuration.entity.ts
+import { BeforeInsert, BeforeUpdate, Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+import { CustomerEntity } from './customer.entity.js';
+import { CredentialProviderEntity } from './credential-provider.entity.js';
+import * as dotenv from 'dotenv';
+
+dotenv.config();
+
+@Entity('providerConfiguration')
+export class ProviderConfigurationEntity {
+	@PrimaryGeneratedColumn('uuid')
+	configId!: string;
+
+	@Column({
+		type: 'varchar',
+		nullable: false,
+	})
+	providerId!: string;
+
+	@Column({
```

Comment:
We can achieve the functionality by storing the encrypted api key in the existing api key table with a new column added for providerId.

Reply:
The api key can be encrypted by the function present already in the api key service file.

Reply:
subaccount, tenantid etc is abstracted inside the defaultSettings column which is json value